src/Mapping/Operator/Simple/LoadDataObject.php

```diff
@@ -74,17 +80,28 @@ public function process($inputData, bool $dryRun = false)
                 $object = DataObject::getById(trim($data));
             } elseif ($this->loadStrategy === self::LOAD_STRATEGY_ATTRIBUTE) {
                 if ($this->attributeName) {
-                    $getter = 'getBy' . $this->attributeName;
-                    $class = ClassDefinition::getById($this->attributeDataObjectClassId);
-                    if (empty($class)) {
-                        throw new InvalidConfigurationException("Class `{$this->attributeDataObjectClassId}` not found.");
-                    }
-                    $className = '\\Pimcore\\Model\\DataObject\\' . ucfirst($class->getName());
-
-                    if ($this->attributeLanguage) {
-                        $object = $className::$getter($data, $this->attributeLanguage, 1);
+                    if ($this->partialMatch) {
+                        $listClassName = $className . '\\Listing';
+                        $listing = new $listClassName();
+                        $listing->setCondition($this->attributeName . ' like ' . $listing->quote("%".$data."%"));
```

Comment:
I would leave adding the `%` to the user, thus the user could define if he wants to search for `$data . '%'` or `'%' . $data` or as full wildcard.

Reply:
Yes, I suppose that would be better. I removed the wildcards from the code. I also added "%" to the example in docs.